app/src/main/java/com/mathewsachin/fategrandautomata/scripts/modules/Support.kt

```diff
@@ -248,13 +248,27 @@ class Support {
         for (preferredServant in preferredServantArray) {
             // Cached pattern. Don't dispose here.
             val pattern = loadSupportImagePattern(preferredServant)
+                .cropFriendLock()
 
             for (servant in Game.SupportListRegion.findAll(pattern)) {
                 yield(servant.Region)
             }
         }
     }
 
+    /**
+     * If you lock your friends, a lock icon shows on the left of servant image,
+     * which can cause matching to fail.
+     * Instead of modifying in-built images and Support Image Maker,
+     * which would need everyone to regenerate their images,
+     * crop out the part which can potentially have the lock.
+     */
+    private fun IPattern.cropFriendLock(): IPattern {
```

Comment:
I feel like this method is a bit dangerous to be declared on IPattern. You don't want to execute this code for any random image, just for servant images. And you also don't want to execute this multiple times in a row.